Feature/docker extensions

TomHAnderson opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

After doing a complete audit of Laminas repositories installed and optionally installed by this repository for extensions those packages use, I've created the necessary `Dockerfile` commands to enable them.

laminas-cache has several supported extensions and all applicable extensions are listed.
laminas-db has many supported extensions and only `pdo-mysql` and `pdo-pgsql` are included.

The `intl` extension is now enabled by default.

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

TomHAnderson committed:
Added all required extensions and many optional extensions for a complete mvc install

Signed-off-by: Tom H Anderson <tom.h.anderson@gmail.com>

weierophinney commented on Dockerfile:
```diff
@@ -22,29 +27,52 @@ RUN mv /var/www/html /var/www/public
 RUN curl -sS https://getcomposer.org/installer \
   | php -- --install-dir=/usr/local/bin --filename=composer
 
+WORKDIR /var/www
```
Why are you moving this _before_ installing extensions, instead of keeping it _after_? Is there a rationale for that?

Generally, I tend to set the workdir immediately prior to commands that need to run inside it.

TomHAnderson replied:
The workdir and installing extensions have no correlation.  I put the workdir where it is as a delimiter between default installation and additional extensions.  

The workdir does two things: 1. set the working directory for the next commands 2. set the working directory when you connect to the container.  I'm using the workdir for #2 just like it was done in the original CLOB.  

If workdir were moved to the bottom of the file then all optional extensions would be moved above apache config.  That is why I felt a delimiter was needed; to better group default and non-default configurations.

TomHAnderson replied:
Workdir has been moved to bottom of Dockerfile

TomHAnderson commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM php:7.3-apache
+FROM php:7.4-apache
```
This was done based on a comment by @Xerkus: https://github.com/laminas/laminas-mvc-skeleton/pull/20#discussion_r450046345

TomHAnderson committed:
Reverted to PHP 7.3

Signed-off-by: Tom H Anderson <tom.h.anderson@gmail.com>

TomHAnderson committed:
Reorganized Dockerfile; workdir now last

Signed-off-by: Tom H Anderson <tom.h.anderson@gmail.com>

weierophinney committed:
qa: combine statements in Dockerfile when possible

While the modular nature of the change suggested are appreciated, each
set of commands to enable a given extension should be done as a single
layer, not multiple layers.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
